src/providers/ldap/sdap_async_groups.c

```diff
@@ -275,8 +275,14 @@ static int sdap_fill_memberships(struct sdap_options *opts,
             hret = HASH_ERROR_KEY_NOT_FOUND;
         } else {
             key.type = HASH_KEY_STRING;
-            key.str = (char *)values[i].data;
+            ret = sss_trim_dn(ctx, (char *)values[i].data, &key.str);
```

Comment:
I'm somewhat worried with additional (alloc+loop+copy) here.
Allocs (without pools) are expensive and processing of those tables might be time consuming (see #5134 as example).

Reply:
This should help, thank you.

But my primary intent was what we discussed over IRC: can we have data in `values[]` trimmed beforehand (to avoid trimming "on the fly")? Or is this impossible?